Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -19,25 +19,72 @@ module.exports.Quantity = class Quantity extends Expression
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration <= otherSmallestDuration
+    else null
 
   sameOrAfter: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value >= parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value >= parseFloat other.value
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration >= otherSmallestDuration
+    else null
 
   after: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value > parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value > parseFloat other.value
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration > otherSmallestDuration
+    else null
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value < parseFloat other.value
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration < otherSmallestDuration
+    else null
     
   equals: (other) ->
     if other.unit? && other.value? then @unit == other.unit && @value == parseFloat other.value else null
 
-time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
+# Hash of time units and their UCUM equivalents
+time_units = {'years': 'a', 'year': 'a'
+  , 'months': 'mo', 'month':'mo'
+  , 'weeks': 'wk', 'week': 'wk'
+  , 'days': 'day', 'day':'d'
+  , 'hours': 'h', 'hour': 'h'
+  , 'minutes': 'min', 'minute': 'min'
+  , 'seconds':'s', 'second':'s'
+  , 'milliseconds' : 'ms', 'millisecond' : 'ms' }
 
 clean_unit = (units) ->
   if time_units[units] then time_units[units] else units
 
+smallestDuration = (qty) ->
+  if parseFloat qty.value
+    millivalue = switch
+      when clean_unit(qty.unit) == 's' then qty.value * 1000
+      when clean_unit(qty.unit) == 'min' then qty.value * 60000
+      when clean_unit(qty.unit) == 'h' then qty.value * 3600000
+      when clean_unit(qty.unit) == 'd' then qty.value * 86400000
+      when clean_unit(qty.unit) == 'wk' then qty.value * 604800000
+      when clean_unit(qty.unit) == 'mo' then qty.value * 2629746000  # Based on a mean month length of 30.436875 days
```

Comment:
I understand why you did it this way, and I don't have a better idea, but... it does make me squirm a bit.  We should probably file a comment on the CQL spec to see if it can identify the approach that should be followed when comparing months to other units.